src/rebase.c

```diff
@@ -270,8 +269,11 @@ static git_rebase *rebase_alloc(const git_rebase_options *rebase_opts)
 		git_rebase_init_options(&rebase->options, GIT_REBASE_OPTIONS_VERSION);
 
 	if (rebase_opts && rebase_opts->rewrite_notes_ref) {
-		if ((rebase->options.rewrite_notes_ref = git__strdup(rebase_opts->rewrite_notes_ref)) == NULL)
+		if ((rebase->options.rewrite_notes_ref =
+			git__strdup(rebase_opts->rewrite_notes_ref)) == NULL) {
+			git__free(rebase);
```

Comment:
We're not in the habit of `free`ing things on allocation failure.  I don't like to have a different pattern here.  If the goal is to silence static code analysis warnings, then we could either:

1. Change this to return an `int` error code so that `GITERR_CHECK_ALLOC` works here, or
2. Silence this another way, perhaps with a macro like `GITERR_CHECK_ALLOC` that returns `NULL` instead of `-1`.

Thoughts?